### ISSUE TITLE
Tests for custom objective/metric with serialization.

### ISSUE THREAD

**Issue by trivialfis** [dmlc/xgboost#9908]
- [ ] Pickle sklearn.
- [ ] Pickle booster.
- [ ] save_model sklearn.
- [ ] save_model booster.
